app/services/record_declarations/base.rb

```diff
@@ -110,5 +111,35 @@ def validate_provider!
       # TODO: Remove the nil? check and fix the test setup so that they build the school cohort, partnership and give us back the actual lead_provider.
       raise ActionController::ParameterMissing, I18n.t(:invalid_participant) unless actual_lead_provider.nil? || lead_provider_from_token == actual_lead_provider
     end
+
+    def parsed_date
+      ActiveRecord::Type::Date.new.cast(declaration_date)
```

Comment:
Does this raise the error below?

Reply:
Good question. I think we should never actually get in here, I will refactor the setup to make sure of that.